src/Whispr.EntityFrameworkCore/Cleaning/OutboxCleanupService.cs

```diff
@@ -44,16 +44,16 @@ private async ValueTask CleanupOutboxMessages(CancellationToken cancellationToke
         {
             var expiredUtc = DateTimeOffset.UtcNow - _retentionPeriod;
             var rowsDeleted = await dbContext.Set<OutboxMessage>()
-                .Take(_batchSize)
                 .OrderBy(x => x.ProcessedAtUtc)
                 .Where(x => x.ProcessedAtUtc < expiredUtc)
+                .Take(_batchSize)
                 .ExecuteDeleteAsync(cancellationToken);
+                
+            logger.LogInformation("Deleted {RowsDeleted} outbox messages older than {RetentionPeriod}", rowsDeleted, _retentionPeriod);
```

Comment:
The if below was previously: `if (rowsDeleted == 0) return`, where the log made sense then, but now we will also log the no-op cases (when rowsDeleted == 0), which I don't really want. Can you add the 0 check bail-out above the log?